utils/Actions.js

```diff
@@ -1,56 +1,134 @@
         });
     }
-    if (attachments) {
-        attachments.forEach((attachment) => {
-            const mediaChannel = guild.channels.find(ch => ch.name === 'media-attachments');
-            if (mediaChannel) mediaChannel.send(new Attachment(attachment.url));
+}
+
+const maybeSendInstagramLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            const match = UrlValidator.matchIGUrl(embed.url);
+            if (match) sendToChannel(message, INSTAGRAM_LINKS, match[0]);
         });
     }
-};
+}
+
+const maybeSendSpotifyLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            const match = UrlValidator.matchSpotifyUrl(embed.url);
+            if (match) sendToChannel(message, SPOTIFY_LINKS, match[0]);
+        });
+    }
+}
+
+const maybeSendVLiveLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            const match = UrlValidator.matchVLiveUrl(embed.url);
+            if (match) sendToChannel(message, VLIVE_LINKS, match[0]);
+        });
+    }
+}
+
+const maybeSendTwitterLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            const match = UrlValidator.matchTwitterUrl(embed.url);
+            if (match) sendToChannel(message, TWITTER_LINKS, match[0]);
+        });
+    }
+}
+
+const maybeSendRedditLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            const match = UrlValidator.matchRedditUrl(embed.url);
+            if (match) sendToChannel(message, REDDIT_LINKS, match[0]);
+        });
+    }
+}
+
+const maybeSendGeneralLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            if (embed.type === 'link') sendToChannel(message, GENERAL_LINKS, embed.url);
+        });
+    }
+}
+
+const maybeSendMediaLinks = (message) => {
+    const embeds = message.embeds;
+    if (embeds) {
+        embeds.forEach(embed => {
+            if (embed.type !== 'rich') sendToChannel(message, MEDIA_LINKS, embed.url);
+        });
+    }
+}
+
+const maybeSendAttachments = (message) => {
+    const attachments = message.attachments;
+    if (attachments) {
+        attachments.forEach(attachment => {
+            sendToChannel(message, MEDIA_ATTACHMENTS, new Attachment(attachment.url), message.content);
+        })
+    }
+}
+
+const sendToChannel = async (originalMessage, channelName, messageToSend, extraContent) => {
+    const guild = originalMessage.guild;
+    const originalChannel = originalMessage.channel;
+    const channel = guild.channels.find(ch => ch.name === channelName);
+    if (channel) {
+        await channel.send(`${DISCORD_BASE_URL}/channels/${guild.id}/${originalChannel.id}/${originalMessage.id}`);
```

Comment:
@iFallUpHill if any of these throws an error, will your setup log them automatically or would we need to catch them ourselves?

Reply:
Apparently bot restarts for uncaught exceptions. We should probably catch the error somewhere and log it.